### DISCUSSION TITLE
Is there a container image to deploy the MLFlow server?

### DISCUSSION THREAD

**Discussion by SayakMukhopadhyay** [mlflow/mlflow#6672] [Q&A] [answered]
I have been looking into running ML Flow on containers and I am unable to find container images provided by the project. The most I have found is some Dockerfiles. Is there no container actually available? In that case, which Dockerfile should I build to get a container which is ready to be run in production?

cc @stsfaroz

**Comment by fzhcary** [comment]
Here is a simple Dockfile,

```
FROM continuumio/miniconda3:4.10.3

RUN pip install mlflow>=1.0 \
    && pip install azure-storage-blob==12.3.0 \
    && pip install numpy==1.21.2 \
    && pip install scipy \
    && pip install pandas==1.3.3 \
    && pip install scikit-learn==0.24.2 \
    && pip install cloudpickle

CMD ["mlflow", "ui", "-h", "0.0.0.0", "-p", "5000"]
```

Just put it in Dockerfile and run it. But remember, mlflow is a library. you need create your own microservice, for example do training, etc.
The above docker file is to run mlflow ui.

**Reply by SayakMukhopadhyay** [reply]
Thanks! I had found some dockerfiles in the repo itself but what I was looking for is an existing container image in dockerhub or somewhere. If that is not available then I will have to create my own containers.

Also, what is the difference between `mlflow ui` and mlflow server`?

**Comment by JHibbard** [comment, marked as answer]
For future posterity, as of [MLflow 1.29.0](https://github.com/mlflow/mlflow/releases/tag/v1.29.0) there's an official Dockerfile for MLflow tracking server! There's no official prebuilt image, but stay tuned as this is planned for MLflow 2.0 release :)
